home_work/home_work1.py

```diff
@@ -0,0 +1,150 @@
+import json
+from pydantic import BaseModel
+from abc import (
+    ABC,
+    abstractmethod
+)
+from datetime import date
+from contextlib import contextmanager
+
+
+class BookModel(BaseModel):
+    title: str
+    author: str
+    created_date: date
+
+
+class JournalModel(BookModel):
+    last_printed_day: date
+
+
+class AbstractReading(ABC):
+    @abstractmethod
+    def reading_info(self):
+        raise NotImplementedError
+
+
+class Book(AbstractReading):
+    def __init__(self, model: BookModel):
+        self._model = model
+
+    def reading_info(self):
+        return f'Book {self._model.title} writed by {self._model.author} from {self._model.created_date}year'
+
+
+class Journal(Book):
+    def __init__(self, model: JournalModel):
+        self._model = model
+
+
+def add_book_notif(func):
+    def wrapper(self, reading):
+        result = func(self, reading)
+        return f'Book{reading._model.title}was added'
+    return wrapper
+
+
+def check_book(func):
+    def wrapper(self, title):
+        for reading in self.list_of_books:
+            if reading._model.title == title:
+                result = func(self, reading)
+                return f'Book {reading._model.title} deleted'
+    return wrapper
+
+
+class Library:
+    def __init__(self):
+        self.list_of_books: list = []
+
+    @add_book_notif
+    def add_material(self, reading):
+        self.list_of_books.append(reading)
+
+    @check_book
+    def del_material(self, reading):
+        self.list_of_books.remove(reading)
```

Comment:
if there will be no book in list it will raise an error, so better to wrap it in try/except

Reply:
Done